File: app/src/main/java/com/ternaryop/photoshelf/drawer/ItemTestDrawerItem.java
package com.ternaryop.photoshelf.drawer;

import android.app.Fragment;
import android.content.Context;
import android.os.Bundle;

import com.ternaryop.photoshelf.Constants;
import com.ternaryop.photoshelf.R;
import com.ternaryop.utils.drawer.adapter.DrawerItem;

/**
 * Created by dave on 28/10/17.
 * Contain info about the item to test
 */

public class ItemTestDrawerItem extends DrawerItem {
    public ItemTestDrawerItem(int itemId, String title, Class<? extends Fragment> fragmentClass) {
        super(itemId, title, fragmentClass);
    }

    public Fragment instantiateFragment(Context context) {
        final Fragment fragment = super.instantiateFragment(context);
        Bundle args = new Bundle();
        args.putString(Constants.EXTRA_URL, context.getString(R.string.test_page_url));
        fragment.setArguments(args);

        return fragment;
    }
}

File: app/src/main/java/com/ternaryop/photoshelf/drawer/ItemTestDrawerItem.kt
package com.ternaryop.photoshelf.drawer

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.ternaryop.photoshelf.EXTRA_URL
import com.ternaryop.photoshelf.R
import com.ternaryop.utils.drawer.adapter.DrawerItem

/**
 * Created by dave on 28/10/17.
 * Contain info about the item to test
 */

class ItemTestDrawerItem(itemId: Int, title: String, fragmentClass: Class<out Fragment>)
    : DrawerItem(itemId, title, fragmentClass) {
    override fun instantiateFragment(context: Context): Fragment {
        val fragment = super.instantiateFragment(context)
        val args = Bundle()
        args.putString(EXTRA_URL, context.getString(R.string.test_page_url))
        fragment.arguments = args

        return fragment
    }
}
